libdnf/comps/group/query.cpp

```diff
@@ -36,10 +35,11 @@ namespace libdnf::comps {
 
 
 GroupQueryWeakPtr GroupQuery::get_weak_ptr() {
-    return GroupQueryWeakPtr(this, &p_impl->data_guard);
+    return GroupQueryWeakPtr(this, &data_guard);
 }
 
-GroupQuery::GroupQuery(const GroupSackWeakPtr & sack) : Query(), sack(sack), p_impl{new Impl()} {
+
+GroupQuery::GroupQuery(const GroupSackWeakPtr & sack) : Query(), sack(sack) {
```

Comment:
You don't need to explicitly call the base class default constructor, you can just leave it out:
```c++
GroupQuery::GroupQuery(const GroupSackWeakPtr & sack) : sack(sack) {
```